### ISSUE TITLE
How to create an empty directory at the /var/lib/ path of the target during the install time

### ISSUE THREAD

**Issue by sunharis** [mesonbuild/meson#7972]
I have a requirement to create an empty directory at the /var/lib/ path of the target during the install time. 

Tried the install_subdir() - did not work. 
Then I used a custom script which is invoked by the add_install_script() api. 

_meson.add_install_script('create_dir.sh', get_option('sharedstatedir'))_

create_dir.sh - 
_#!/bin/sh
mkdir -p ${DESTDIR}/$1/path_

This did not work either

I found that only after appending the prefix path , the directory gets created in the target under the /usr path.  For that the custom script had - 
_#!/bin/sh
mkdir "${DESTDIR}/${MESON_INSTALL_PREFIX}/path_

How i can get the directory created under the /var/lib at the target machine?

~

**Comment by sunharis** [comment]
@dcbaker @jpakkane Can you please check this ?

**Comment by eli-schwartz** [comment]
Duplicate of: #2904 #1604

It's currently impossible, the install script is your only option.

```meson
path = join_paths(get_option('localstatedir'), 'lib/myproject')
meson.add_install_script('sh', '-c', 'mkdir -p "$DESTDIR/$1"', '_', path)
```

**Comment by sunharis** [comment]
Thanks @eli-schwartz for responding. I tried the meson instructions as you mentioned.

After the meson build, i do a bitbake to build the image. Once i install this image in the target, i see that the directory _myproject_ is not created under root path ("/"). Instead it creates the  directory under read only file system as below.

_./media/rofs-9f988762/var/lib/myproject_

I expect the directory created at root path as _/var/lib/myproject_

**Comment by eli-schwartz** [comment]
Is ./media/rofs-9f988762 your DESTDIR install root? Then that's supposed to happen... then the packaged install image will get merged later onto the target device's / directory.

**Comment by sunharis** [comment]
Yes. This is the DESTDIR.  But, at the target, i should get the directory created under root path. 
I observed that only when i tag the path with MESON_INSTALL_PREFIX, the directory goes to the root path , but that is created under _/usr_

**Comment by sunharis** [comment]
> packaged install image will get merged later onto the target device's / directory.
I think this is not happening if i ask meson to create a directory without tagging it with MESON_INSTALL_PREFIX

**Comment by eli-schwartz** [comment]
Well, as far as I can tell, there's literally nothing wrong here. You said `DESTDIR=./media/rofs-9f988762 ninja install` creates `./media/rofs-9f988762/var/lib/myproject`

Which is precisely what you asked for.

**Comment by eli-schwartz** [comment]
Duplicate of #1604
